### PR TITLE
Add post on beta API and migration guidelines

### PR DESCRIPTION
**Pull request by qu1queee**
# Changes

Add blog post on beta API

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
Add blog post with migration guidelines for beta API.
```

### CODE REVIEW

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+title: "Introducing Shipwright Beta API"
+date: 2023-11-07T16:00:00-04:00
+draft: false
+author: "Enrique Encalada ([@qu1queee](https://github.com/qu1queee))"
+---
+
+About a year ago, we published a blog [post](https://shipwright.io/blog/2022/10/25/bringing-shipwright-to-beta-and-beyond/)
+in which we outline our vision and our values. Part of this vision was to advance
+our API to enhance simplicity, consistency and signalize a higher level of maturity.
```
Comment (rolfedh) [inline content/en/blog/posts/2023-11-07-beta-api.md:10]:
```suggestion
our API to enhance its simplicity and consistency, and to signal a higher level of maturity.
```

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Today, as part of our release _v0.12.0_, we are introducing our _beta_ API. 
+The _beta_ API brings multiple changes, as a result of accumulated experience operating
+the _alpha_ API and incorporating valuable user feedback.
+
+With the introduction of the _beta_ API, users can have confidence that our
+core components have been battle tested and that using our different features is
```
Comment (rolfedh) [inline content/en/blog/posts/2023-11-07-beta-api.md:17]:
```suggestion
core components have been battle-tested, and using our different features is
```

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+draft: false
+author: "Enrique Encalada ([@qu1queee](https://github.com/qu1queee))"
+---
+
+About a year ago, we published a blog [post](https://shipwright.io/blog/2022/10/25/bringing-shipwright-to-beta-and-beyond/)
+in which we outline our vision and our values. Part of this vision was to advance
+our API to enhance simplicity, consistency and signalize a higher level of maturity.
+
+Today, as part of our release _v0.12.0_, we are introducing our _beta_ API. 
+The _beta_ API brings multiple changes, as a result of accumulated experience operating
```
Comment (rolfedh) [inline content/en/blog/posts/2023-11-07-beta-api.md:13]:
```suggestion
The _beta_ API brings multiple changes as a result of accumulated experience operating
```

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+considered a safe practice.
+
+_We want to thank our community for their contributions and support in redefining this new API!_
+
+## Beta API
+
+The _beta_ API is available starting from the _v0.12.0_ release. The release is available 
+across our [cli](https://github.com/shipwright-io/cli), [operator](https://github.com/shipwright-io/operator) and [build](https://github.com/shipwright-io/build) repository.
+
+Within the _v0.12.0_ release, a conversion webhook has been introduced to ensure backwards
```
Comment (rolfedh) [inline content/en/blog/posts/2023-11-07-beta-api.md:27]:
```suggestion
Within the _v0.12.0_ release, a conversion webhook has been introduced to ensure backward
```

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+|  Old field                       |                            New field                                     |
+|----------------------------------|:------------------------------------------------------------------------:|
+| `.spec.source.url`               |  `.spec.source.git.url`                                                  |
+| `.spec.source.bundleContainer`   |  `.spec.source.ociArtifact`                                              |
+| `.spec.sources` for `LocalCopy`  |  `.spec.source.local`                                                    |
+| `.spec.source.credentials`       |  `.spec.source.git.cloneSecret` or `.spec.source.ociArtifact.pullSecret` |
+| `.spec.output.credentials`       |  `.spec.output.pushSecret`                                               |
+
+
+See example for Git source type:
```
Comment (rolfedh) [inline content/en/blog/posts/2023-11-07-beta-api.md:60]:
```suggestion
See this example of the Git source type:
```

**Review by rolfedh — COMMENTED** [review]
Hi @qu1queee. Nice work again! I've offered some minor suggestions.

**Comment by qu1queee** [comment]
thanks @rolfedh !

**Review by rolfedh — APPROVED** [review]
LGTM!

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+
+### Changes to BuildRun API fields
+
+|  Old field                  |                 New field              |
+|-----------------------------|:--------------------------------------:|
+| `.spec.buildSpec`           | `.spec.build.spec`                     |
+| `.spec.buildRef`            | `.spec.build.name`                     |
+| `.spec.sources`             | `.spec.source` only for `Local`        |
+| `.spec.serviceAccount.name` | `.spec.serviceAccount` with `.generate`|
```
Comment (adambkaplan) [inline content/en/blog/posts/2023-11-07-beta-api.md:106]: I think the intent here is to state that `.spec.serviceAccount.generate` is gone?

```suggestion
| `.spec.serviceAccount.generate` | `.spec.serviceAccount` with `.generate`|

_Note: generated service accounts is a deprecated feature, and may be removed in a future release._
```

File: content/en/blog/posts/2023-11-07-beta-api.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+The _beta_ API is available starting from the _v0.12.0_ release. The release is available 
+across our [cli](https://github.com/shipwright-io/cli), [operator](https://github.com/shipwright-io/operator) and [build](https://github.com/shipwright-io/build) repository.
+
+Within the _v0.12.0_ release, a conversion webhook has been introduced to ensure backward
+compatibility between the _v1alpha1_ and _v1beta1_. 
+
+We strongly encourage both current and future users to adopt the _beta_ API to benefit from
+its enhanced definition.
+
```
Comment (adambkaplan) [inline content/en/blog/posts/2023-11-07-beta-api.md:32]: Do we have any other notable new features? Deprecations?

**Review by adambkaplan — COMMENTED** [review]
/approve

Looks great! The generated service accounts item got me thinking about new features and deprecations. It's been a while - anything worth reporting beyond the beta API?

File: content/en/blog/posts/2023-11-07-beta-api.md (hunk 6 above)
Comment (SaschaSchwarze0) [reply content/en/blog/posts/2023-11-07-beta-api.md:106]: I like that additional sentence.

File: content/en/blog/posts/2023-11-07-beta-api.md (hunk 7 above)
Comment (qu1queee) [reply content/en/blog/posts/2023-11-07-beta-api.md:32]: very few, like improvements in terms of security for the pods running on the `shipwright-build` namespace. I think is fine not adding them here, as this one focus more on the beta API only.

**Review by SaschaSchwarze0 — APPROVED** [review]
/lgtm

**Pull request merged**